router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/utils.go

```diff
@@ -0,0 +1,278 @@
+
+			statusCode, category, errorMessage = handleMarketoErrorCode(errorCode)
+			errorMessage = fmt.Sprintf("%s: %s", message, errorMessage)
+		} else {
+			// Handle the case where Errors array is empty
+			statusCode, category, errorMessage = handleMarketoErrorCode("Unknown")
+		}
+
+	}
+
+	return statusCode, category, errorMessage
+}
+
+// ==== Response Parsing End ====
+
+func createCSVFile(destinationID string, destConfig MarketoConfig, input []common.AsyncJob, dataHashToJobId map[string]int64) (string, []string, []int64, []int64, error) {
+	csvFilePath := fmt.Sprintf("/tmp/%s_%s.csv", destinationID, "marketo_bulk_upload")
+	csvFile, err := os.Create(csvFilePath)
+	if err != nil {
+		return "", nil, nil, nil, err
+	}
+	defer csvFile.Close()
+
+	overflowedJobIDs := make([]int64, 0)
+	insertedJobIDs := make([]int64, 0)
+	currentSize := int64(0)
+
+	headers := make(map[string]int)
+	var headerOrder []string
+
+	// Create a CSV writer
+	writer := csv.NewWriter(csvFile)
+	defer writer.Flush()
+
+	// First pass: collect all unique headers we are taking value as its the marketo field name
+	for _, value := range destConfig.FieldsMapping {
+		if _, exists := headers[value]; !exists {
+			headers[value] = len(headerOrder)
+			headerOrder = append(headerOrder, value)
+		}
+	}
+
+	// Write header row
+	headerRow := strings.Join(headerOrder, ",") + "\n"
+
+	err = writer.Write(headerOrder)
+	if err != nil {
+		return "", nil, nil, nil, err
+	}
+	currentSize += int64(len(headerRow))
+
+	// Second pass: write data rows
+	for _, job := range input {
+		message := job.Message
+		row := make([]string, len(headerOrder))
+		for key, value := range message {
+			if _, exists := headers[key]; exists {
+				row[headers[key]] = fmt.Sprintf("%v", value)
+			}
+		}
+		line := strings.Join(row, ",") + "\n"
+		lineSize := int64(len(line))
+
+		if currentSize+lineSize > 10*1024*1024 { // 10MB in bytes
+			overflowedJobIDs = append(overflowedJobIDs, int64(job.Metadata["job_id"].(float64)))
+			continue
+		}
+
+		err := writer.Write(row)
+		if err != nil {
+			return "", nil, nil, nil, err
+		}
+		currentSize += lineSize
+		insertedJobIDs = append(insertedJobIDs, int64(job.Metadata["job_id"].(float64)))
+		// Calculate hash code for the row
+		dataHash := calculateHashCode(row)
+		// Store the mapping of data hash to job ID
+		dataHashToJobId[dataHash] = int64(job.Metadata["job_id"].(float64))
+	}
+
+	return csvFilePath, headerOrder, insertedJobIDs, overflowedJobIDs, nil
+}
+
+func calculateHashCode(data []string) string {
+	// Join the strings into a single string with a separator
+	joinedStrings := strings.Join(data, ",")
+	hash := sha256.New()
+	hash.Write([]byte(joinedStrings))
+	hashBytes := hash.Sum(nil)
+	hashCode := fmt.Sprintf("%x", hashBytes)
+
+	return hashCode
+}
+
+func sendHTTPRequest(uploadURL, csvFilePath string, accessToken string, deduplicationField string) (*http.Response, error) {
+	file, err := os.Open(csvFilePath)
+	if err != nil {
+		return nil, err
```

Comment:
Apart from returning error only, we can add oneliner as follows:
`fmt.Errorf("error while opening the file: %v", err)`

Reply:
This fixed